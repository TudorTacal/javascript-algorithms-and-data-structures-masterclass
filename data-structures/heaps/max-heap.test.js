const { MaxHeap } = require('./max-heap');

describe('maxHeap', () => {
  test('#insert should insert a value in the correct place', () => {
    let maxHeap = new MaxHeap();

    maxHeap.insert(41);
    maxHeap.insert(39);
    maxHeap.insert(33);
    maxHeap.insert(18);
    maxHeap.insert(27);
    maxHeap.insert(12);
    maxHeap.insert(55);



    //        55
    //    39      41
    // 18   27 12   33

    expect(maxHeap).toMatchSnapshot();
  });
});