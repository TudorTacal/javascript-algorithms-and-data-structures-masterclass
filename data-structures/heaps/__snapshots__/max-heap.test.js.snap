// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maxHeap #insert should insert a value in the correct place 1`] = `
MaxHeap {
  "values": Array [
    55,
    39,
    41,
    18,
    27,
    12,
    33,
  ],
}
`;
